staging/src/k8s.io/client-go/applyconfigurations/admissionregistration/v1/mutatingwebhook.go

```diff
@@ -0,0 +1,256 @@
+	b.fields.ObjectSelector = value
+	return b
+}
+
+// ObjectSelector gets the ObjectSelector field from the declarative configuration.
+func (b *MutatingWebhookApplyConfiguration) ObjectSelector() (value *metav1.LabelSelectorApplyConfiguration, ok bool) {
+	return b.fields.ObjectSelector, b.fields.ObjectSelector != nil
+}
+
+// SetSideEffects sets the SideEffects field in the declarative configuration to the given value.
+func (b *MutatingWebhookApplyConfiguration) SetSideEffects(value admissionregistrationv1.SideEffectClass) *MutatingWebhookApplyConfiguration {
+	b.fields.SideEffects = &value
+	return b
+}
+
+// SideEffects gets the SideEffects field from the declarative configuration.
+func (b *MutatingWebhookApplyConfiguration) SideEffects() (value admissionregistrationv1.SideEffectClass, ok bool) {
+	if v := b.fields.SideEffects; v != nil {
+		return *v, true
+	}
+	return value, false
+}
+
+// SetTimeoutSeconds sets the TimeoutSeconds field in the declarative configuration to the given value.
+func (b *MutatingWebhookApplyConfiguration) SetTimeoutSeconds(value int32) *MutatingWebhookApplyConfiguration {
+	b.fields.TimeoutSeconds = &value
+	return b
+}
+
+// TimeoutSeconds gets the TimeoutSeconds field from the declarative configuration.
+func (b *MutatingWebhookApplyConfiguration) TimeoutSeconds() (value int32, ok bool) {
+	if v := b.fields.TimeoutSeconds; v != nil {
+		return *v, true
+	}
+	return value, false
+}
+
+// SetAdmissionReviewVersions sets the AdmissionReviewVersions field in the declarative configuration to the given value.
+func (b *MutatingWebhookApplyConfiguration) SetAdmissionReviewVersions(value []string) *MutatingWebhookApplyConfiguration {
+	b.fields.AdmissionReviewVersions = &value
+	return b
+}
+
+// AdmissionReviewVersions gets the AdmissionReviewVersions field from the declarative configuration.
+func (b *MutatingWebhookApplyConfiguration) AdmissionReviewVersions() (value []string, ok bool) {
+	if v := b.fields.AdmissionReviewVersions; v != nil {
+		return *v, true
+	}
+	return value, false
+}
+
+// SetReinvocationPolicy sets the ReinvocationPolicy field in the declarative configuration to the given value.
+func (b *MutatingWebhookApplyConfiguration) SetReinvocationPolicy(value admissionregistrationv1.ReinvocationPolicyType) *MutatingWebhookApplyConfiguration {
+	b.fields.ReinvocationPolicy = &value
+	return b
+}
+
+// ReinvocationPolicy gets the ReinvocationPolicy field from the declarative configuration.
+func (b *MutatingWebhookApplyConfiguration) ReinvocationPolicy() (value admissionregistrationv1.ReinvocationPolicyType, ok bool) {
+	if v := b.fields.ReinvocationPolicy; v != nil {
+		return *v, true
+	}
+	return value, false
+}
+
+// ToUnstructured converts MutatingWebhookApplyConfiguration to unstructured.
+func (b *MutatingWebhookApplyConfiguration) ToUnstructured() interface{} {
+	if b == nil {
+		return nil
+	}
+	b.preMarshal()
```

Comment:
What's the expected goal of `preMarshal/postUnmarshal`?
Also, why do we need to hide the fields behind `b.fields`. I think my question is, why can't we just remove this function and ask people to do themselves:
```golang
runtime.DefaultUnstructuredConvereter.ToUnstructured(&MyMutatingWebhookApplyConfiguration)
```

Reply:
Correct me if I'm wrong @jpbetz but by using the builders instead of the structs approach, the convention is to set struct fields as private. Since the fields are public by default and directly changing them to private (renaming) also causes problems when converting to unstructured, keeping the fields as public and hiding them within a private substruct was a simpler solution.

Reply:
Since I've exposed the structs (see https://github.com/kubernetes/kubernetes/pull/99012#discussion_r577944856), we no longer have preMarshal/postUnmarshal, but yes, those functions were added to deal with some of the problems with serialization that come up when keeping the fields are kept unexposed. The most problematic part is that inlined types from other packages are difficult to support.